src/librustc_borrowck/borrowck/mod.rs

```diff
@@ -236,6 +296,47 @@ impl<'tcx> Loan<'tcx> {
     pub fn loan_path(&self) -> Rc<LoanPath<'tcx>> {
         self.loan_path.clone()
     }
+
+    pub fn kind(&self) -> ty::BorrowKind {
+        self.kind
+    }
+
+    pub fn span(&self) -> Span {
+        self.span
+    }
+
+    pub fn span_in_def(&self, def_id: DefId, tcx: TyCtxt) -> Option<Span> {
```

Comment:
The reason the helper is here is that `gen_scope` and `kill_scope` are private to `Loan`. I had previously made public getters for them, though this seemed cleaner because it exposed less of `Loan`'s internals. I didn't like having this logic here though since it was so specific to the visualization logic, but it also felt like something the `Loan` should be responsible for given the use of the private fields.